doc/naming.md

```diff
@@ -0,0 +1,50 @@
+#gRPC Naming and Discovery Support
+
+## Overview
+
+gRPC supports DNS as the default name-system. A number of alternative name-systems are used in various deployments. We propose an API that is general enough to support a range of name-systems and the corresponding syntax for names. The gRPC client library in various languages will provide a plugin mechanism so resolvers for different name-systems can be plugged in.
+
+## Detailed Proposal
+
+ A fully qualified, self contained name used for gRPC channel construction uses the syntax:
+
+```scheme://authority/endpoint_name
```

Comment:
This shows up empty. The scheme should be on the next line. The things on the same like after the \`\`\` is the syntax to use for the syntax highlighter, like "java".